adapters/Unity/UnityAdapter/GADMAdapterUnity.m

```diff
@@ -13,220 +13,185 @@
-      [strongNetworkConnector adapter:self didFailAd:error];
-    }
-  }
 }
 
 - (void)unityAdsDidFinish:(NSString *)placementID withFinishState:(UnityAdsFinishState)state {
-  if ([placementID isEqualToString:_placementID]) {
-    id<GADMAdNetworkConnector> strongNetworkConnector = _networkConnector;
-    if (strongNetworkConnector) {
-      [strongNetworkConnector adapterDidDismissInterstitial:self];
+    if ([placementID isEqualToString:_placementID]) {
+        id<GADMAdNetworkConnector> strongNetworkConnector = _networkConnector;
+        if (strongNetworkConnector) {
+            [strongNetworkConnector adapterDidDismissInterstitial:self];
+        }
     }
-  }
 }
 
 - (void)unityAdsReady:(NSString *)placementID {
-  id<GADMAdNetworkConnector> strongNetworkConnector = _networkConnector;
-  if (strongNetworkConnector && [placementID isEqualToString:_placementID]) {
-    [strongNetworkConnector adapterDidReceiveInterstitial:self];
-  }
 }
 
 - (void)unityAdsDidClick:(NSString *)placementID {
-  id<GADMAdNetworkConnector> strongNetworkConnector = _networkConnector;
-  // The Unity Ads SDK doesn't provide an event for leaving the application, so the adapter assumes
-  // that a click event indicates the user is leaving the application for a browser or deeplink, and
-  // notifies the Google Mobile Ads SDK accordingly.
-  if (strongNetworkConnector && [placementID isEqualToString:_placementID]) {
-    [strongNetworkConnector adapterDidGetAdClick:self];
-    [strongNetworkConnector adapterWillLeaveApplication:self];
-  }
+    id<GADMAdNetworkConnector> strongNetworkConnector = _networkConnector;
+    // The Unity Ads SDK doesn't provide an event for leaving the application, so the adapter assumes
+    // that a click event indicates the user is leaving the application for a browser or deeplink, and
+    // notifies the Google Mobile Ads SDK accordingly.
+    if (strongNetworkConnector && [placementID isEqualToString:_placementID]) {
+        [strongNetworkConnector adapterDidGetAdClick:self];
+        [strongNetworkConnector adapterWillLeaveApplication:self];
+    }
 }
 
 - (void)unityAdsDidError:(UnityAdsError)error withMessage:(NSString *)message {
-  id<GADMAdNetworkConnector> strongNetworkConnector = _networkConnector;
-  if (strongNetworkConnector) {
-    [strongNetworkConnector adapterWillDismissInterstitial:self];
-    [strongNetworkConnector adapterDidDismissInterstitial:self];
-  }
+    id<GADMAdNetworkConnector> strongNetworkConnector = _networkConnector;
+    if (strongNetworkConnector) {
+        [strongNetworkConnector adapterWillDismissInterstitial:self];
+        [strongNetworkConnector adapterDidDismissInterstitial:self];
+    }
 }
 
 - (void)unityAdsDidStart:(nonnull NSString *)placementID {
-  // nothing to do
+    // nothing to do
+}
+
+
+// UnityAdsInitialization Delegate methods
+- (void)initializationComplete {
+    NSLog(@"Unity Ads initialized successfully");
```

Comment:
we need to tell AdMob that SDK is initialized through callback.

Reply:
@serge-14 How do I do this? We are given a GADMAdNetworkConnector in initWithGADMAdNetworkConnector which is used below to notify failure, but there is no such method for initialization success

Reply:
but do we need this method? I think we should do more than just print log...